girder_annotation/girder_large_image_annotation/rest/annotation.py

```diff
@@ -558,6 +561,94 @@ def deleteItemAnnotations(self, item):
+        accessPipeline = [
+            {'$match': {
+                '$or': [
+                    {'access.users':
+                        {'$elemMatch': {
+                            'id': user['_id'],
+                            'level': {'$gte': 2}
+                        }}},
+                    {'access.groups':
+                        {'$elemMatch': {
+                            'id': {'$in': user['groups']},
+                            'level': {'$gte': 2}
+                        }}}
+                ]
+            }}
+        ] if not user['admin'] else []
+        pipeline = [
+            {'$match': {'_id': 'none'}},
+            {'$unionWith': {
+                'coll': 'folder',
+                'pipeline': [{'$match': {'_id': ObjectId(id)}}] +
+                recursivePipeline +
+                [{'$unionWith': {
+                    'coll': 'folder',
+                    'pipeline': [{'$match': {'_id': ObjectId(id)}}]
+                }}, {'$lookup': {
+                    'from': 'item',
+                    'localField': '_id',
+                    'foreignField': 'folderId',
+                    'as': '__items'
+                }}, {'$lookup': {
+                    'from': 'annotation',
+                    'localField': '__items._id',
+                    'foreignField': 'itemId',
+                    'as': '__annotations'
+                }}, {'$unwind': '$__annotations'},
+                    {'$replaceRoot': {'newRoot': '$__annotations'}},
+                    {'$match': {'_active': {'$ne': False}}}
+                ] + accessPipeline
+            }},
+        ]
+        pipeline = pipeline + [{'$sort': {sort: sortDir}}] if sort else pipeline
+        pipeline = pipeline + [{'$skip': offset}] if offset else pipeline
+        pipeline = pipeline + [{'$limit': limit}] if limit else pipeline
+
+        return Annotation().collection.aggregate(pipeline)
+
+    @autoDescribeRoute(
+        Description('Check if there are any annotations from the items in a folder')
+        .param('id', 'The ID of the folder', required=True, paramType='path')
+        .param('recurse', 'Whether or not to recursively check '
+               'subfolders for annotations', required=False, default=True, dataType='boolean')
+        .errorResponse()
+    )
+    @access.public
+    def existFolderAnnotations(self, id, recurse):
+        annotations = self.getFolderAnnotations(id, recurse, self.getCurrentUser(), 1)
+        try:
+            next(annotations)
+            yield True
+        except StopIteration:
+            yield False
+
+    @autoDescribeRoute(
+        Description('Get the annotations from the items in a folder')
+        .param('id', 'The ID of the folder', required=True, paramType='path')
+        .param('recurse', 'Whether or not to retrieve all '
+               'annotations from subfolders', required=False, default=False, dataType='boolean')
```

Comment:
Why did you choose to have recurse default to false for this endpoint (and true on the present endpoint)?

Reply:
My thought was that when you think of an endpoint that gets annotations from a folder someone is more likely to assume that it's just returning the annotations from that folder and not all other folders it contains as well.  Whereas with the present endpoint I thought they'd be more likely to use it when trying to determine if a user owns any annotations in the folder or its subfolders.  But I realize this logic isn't very intuitive, so I could change it so they're both the same.